File: interview-coach-frontend/src/types/resumeVersion.ts
export interface ResumeVersionSummary {
  id: string;
  version_name: string;
  file_name: string;
  resume_score: number;
  created_at: string;
}

export interface ResumeVersionListResponse {
  total: number;
  items: ResumeVersionSummary[];
}

export interface ResumeVersionDetail {
  id: string;
  version_name: string;
  file_name: string;
  resume_score: number;
  skills: string[];
  extracted_text: string;
  created_at: string;
}

export interface CompareResumeVersionsRequest {
  old_version_id: string;
  new_version_id: string;
}

export interface CompareResumeVersionsResponse {
  old_version_id: string;
  new_version_id: string;
  old_score: number;
  new_score: number;
  score_change: number;
  added_skills: string[];
  removed_skills: string[];
  new_recommendations: string[];
}

File: interview-coach-backend/schemas/resume_version.py
from datetime import datetime
from pydantic import BaseModel, Field


class SaveResumeVersionRequest(BaseModel):
    version_name: str = Field(..., min_length=1, max_length=255)


class ResumeVersionSummary(BaseModel):
    id: str
    version_name: str
    file_name: str
    resume_score: float
    created_at: datetime

    class Config:
        from_attributes = True


class ResumeVersionListResponse(BaseModel):
    total: int
    items: list[ResumeVersionSummary]


class ResumeVersionDetail(BaseModel):
    id: str
    version_name: str
    file_name: str
    resume_score: float
    skills: list[str]
    extracted_text: str
    created_at: datetime


class CompareResumeVersionsRequest(BaseModel):
    old_version_id: str
    new_version_id: str


class CompareResumeVersionsResponse(BaseModel):
    old_version_id: str
    new_version_id: str
    old_score: float
    new_score: float
    score_change: float
    added_skills: list[str]
    removed_skills: list[str]
    new_recommendations: list[str]